### PR TITLE
add a cmake file for yaml-cpp

### PR DESCRIPTION
**Pull request by vrabaud**

### CODE REVIEW

File: cmake/Modules/FindYAML_CPP.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+# Cache variables:
+#
+# - YAML_CPP_LIBRARIES
+# - YAML_CPP_INCLUDE_DIR (as done upstream for now)
+# - YAML_CPP_INCLUDE_DIRS (as done usually)
+#
+# Output Variables on Linux only for now:
+#
+# - YAML_CPP_FOUND: Boolean that indicates if the package was found.
```
Comment (wjwwood) [inline cmake/Modules/FindYAML_CPP.cmake:16]: This must be set on all platforms.

**Comment by vrabaud** [comment]
ok, I updated but I am not sure it is worth it anymore if the default Linux one works on OSX.

**Comment by wjwwood** [comment]
@vrabaud what did you change? Looks like `YAML_CPP_FOUND` is still not set in all cases.

> ok, I updated but I am not sure it is worth it anymore if the default Linux one works on OSX.

Should we close this then? Why was it originally opened?

**Comment by vrabaud** [comment]
YAML_CPP_FOUND is set on the Apple side and automatically on Linux with pkg_check_modules.

If you can give me confirmation that pkg_check_modules works for you on Mac and that the variables are correct, I'll close this PR.

File: cmake/Modules/FindYAML_CPP.cmake [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# - YAML_CPP_VERSION: major.minor.patch yaml cpp version string
+
+if (APPLE)
+  #find yaml cpp for mac os
+  find_library(YAML_CPP_LIBRARIES NAMES yaml-cpp)
+  find_path(YAML_CPP_H_INCLUDE_DIR yaml-cpp/yaml.h )
+  set(YAML_CPP_INCLUDE_DIRS ${YAML_CPP_H_INCLUDE_DIR}/yaml-cpp)
+  if (YAML_CPP_LIBRARIES)
```
Comment (wjwwood) [inline cmake/Modules/FindYAML_CPP.cmake:24]: If this is not true, then `YAML_CPP_FOUND` is never set. It should have an else setting it to `0` or it should be set to `NOTFOUND` or whatever before doing this conditional.

**Comment by wjwwood** [comment]
I can use `pkg-config` to get all the needed stuff for `yaml-cpp` on OS X 10.10: https://github.com/ros-perception/image_common/pull/31#issuecomment-85784844

**Comment by vrabaud** [comment]
ok, closing then. Thx

**Pull request closed without merging**